src/sha2.py

```diff
@@ -20,22 +20,119 @@
+        # print(f'{w + delta_a:08x} {w + delta_e:08x}')
+        return res, sample, delta_a, delta_e
+
+    def compress(self, w, iv=None, trace_size=2):
+        first_block = iv is None
+        if first_block:
+            iv = self.iv
+        if len(w.shape) == 1:
+            state = iv.copy()
+        else:
+            state = np.transpose(np.broadcast_to(iv, (w.shape[1], 8)))
+        round_count = len(self.round_const)
+        trace = []
+        for i in range(round_count):
+            if i >= 16:
+                wlast = self.sigma1(w[14]) + w[9] + self.sigma0(w[1]) + w[0]
+            else:
+                wlast = w[0].copy()
+            w[:-1] = w[1:]
+            w[-1] = wlast
+            state, sample, junk, junk = self.round(
+                state,
+                wlast,
+                self.round_const[i],
+                i < trace_size
+            )
+            if i < trace_size:
+                trace.append(sample)
+        if first_block:
+            state += iv
+            junk, junk, delta_a, delta_e = self.round(
+                state,
+                0,
+                self.round_const[0],
+                False
+            )
+            return np.append(state, [delta_a, delta_e])
+        return (
+            np.transpose(np.broadcast_to(iv, state.shape[::-1])) + state,
+            np.array(trace)
+        )
+
     @staticmethod
     def show(data, size, total_nibble_count):
         if size == -1:
             return '.' * total_nibble_count
         nibble_count = (size >> 2) + 1
-        formatter = '.' * (total_nibble_count - nibble_count) + '{{:0{}x}}'.format(nibble_count)
+        formatter = '.' * (total_nibble_count - nibble_count) + \
+                    '{{:0{}x}}'.format(nibble_count)
 
         return formatter.format(data)
 
 
-class Sha256:
+class Sha256(Sha2):
     dtype = np.uint32
     bit_count = 32
     nibble_count = 8
     formatter = '{:08x} '
-    round_const = (np.uint32(0x428a2f98), np.uint32(0x71374491))
+    round_const = [np.uint32(x) for x in [
+        0x428a2f98, 0x71374491, 0xb5c0fbcf, 0xe9b5dba5,
+        0x3956c25b, 0x59f111f1, 0x923f82a4, 0xab1c5ed5,
+        0xd807aa98, 0x12835b01, 0x243185be, 0x550c7dc3,
+        0x72be5d74, 0x80deb1fe, 0x9bdc06a7, 0xc19bf174,
+        0xe49b69c1, 0xefbe4786, 0x0fc19dc6, 0x240ca1cc,
+        0x2de92c6f, 0x4a7484aa, 0x5cb0a9dc, 0x76f988da,
+        0x983e5152, 0xa831c66d, 0xb00327c8, 0xbf597fc7,
+        0xc6e00bf3, 0xd5a79147, 0x06ca6351, 0x14292967,
+        0x27b70a85, 0x2e1b2138, 0x4d2c6dfc, 0x53380d13,
+        0x650a7354, 0x766a0abb, 0x81c2c92e, 0x92722c85,
+        0xa2bfe8a1, 0xa81a664b, 0xc24b8b70, 0xc76c51a3,
+        0xd192e819, 0xd6990624, 0xf40e3585, 0x106aa070,
+        0x19a4c116, 0x1e376c08, 0x2748774c, 0x34b0bcb5,
+        0x391c0cb3, 0x4ed8aa4a, 0x5b9cca4f, 0x682e6ff3,
+        0x748f82ee, 0x78a5636f, 0x84c87814, 0x8cc70208,
+        0x90befffa, 0xa4506ceb, 0xbef9a3f7, 0xc67178f2,
+    ]]
```

Comment:
```suggestion
    round_const = np.array([
        0x428a2f98, 0x71374491, 0xb5c0fbcf, 0xe9b5dba5,
        0x3956c25b, 0x59f111f1, 0x923f82a4, 0xab1c5ed5,
        0xd807aa98, 0x12835b01, 0x243185be, 0x550c7dc3,
        0x72be5d74, 0x80deb1fe, 0x9bdc06a7, 0xc19bf174,
        0xe49b69c1, 0xefbe4786, 0x0fc19dc6, 0x240ca1cc,
        0x2de92c6f, 0x4a7484aa, 0x5cb0a9dc, 0x76f988da,
        0x983e5152, 0xa831c66d, 0xb00327c8, 0xbf597fc7,
        0xc6e00bf3, 0xd5a79147, 0x06ca6351, 0x14292967,
        0x27b70a85, 0x2e1b2138, 0x4d2c6dfc, 0x53380d13,
        0x650a7354, 0x766a0abb, 0x81c2c92e, 0x92722c85,
        0xa2bfe8a1, 0xa81a664b, 0xc24b8b70, 0xc76c51a3,
        0xd192e819, 0xd6990624, 0xf40e3585, 0x106aa070,
        0x19a4c116, 0x1e376c08, 0x2748774c, 0x34b0bcb5,
        0x391c0cb3, 0x4ed8aa4a, 0x5b9cca4f, 0x682e6ff3,
        0x748f82ee, 0x78a5636f, 0x84c87814, 0x8cc70208,
        0x90befffa, 0xa4506ceb, 0xbef9a3f7, 0xc67178f2,
    ], dtype=np.uint32)
```
Just to simplify and be consistent with the other constant arrays